config/routes.rb

```diff
@@ -3,4 +3,31 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  # CREATE Tweets
+  resources :tweets, only: [:new, :create]
+   # UPDATE Tweets
+  resources :tweets, only: [:edit, :update]
+
+  resources :tweets do
+    member do
+      post 'like', to: 'tweets#like'
+      delete 'unlike', to: 'tweets#unlike'
+      post 'retweet', to: 'tweets#retweet'
+      post 'quote', to: 'tweets#quote'
+      get 'reply', to: 'tweets#reply'
+      post 'bookmark', to: 'tweets#bookmark'
+      get 'stats', to: 'tweets#stats'
+    end
+  end
```

Comment:
Is not necessary to declare it separately  yo can declare all this routes in a single block like this

```suggestion
  resources :tweets, only: [:edit, :update, :new, :create] do
    resources :
    member do
      post 'like', to: 'tweets#like'
      delete 'unlike', to: 'tweets#unlike'
      post 'retweet', to: 'tweets#retweet'
      post 'quote', to: 'tweets#quote'
      get 'reply', to: 'tweets#reply'
      post 'bookmark', to: 'tweets#bookmark'
      get 'stats', to: 'tweets#stats'
    end
  end
```